docs: db routing with static embedding (#63999)
metabase/metabase: docs/permissions/database-routing.md

Database routing is useful for:

- Managing interactive embedding setups where each customer has their own database with identical schemas.

>   Database routing can't be used with static embedding setups, because people using staticly embedded questions or dashboard aren't authenticated into Metabase, and so Metabase can't know who is viewing the data and can't route the queries.

Reviewer: ```suggestion
  Database routing can't be used with static embedding, because database routing requires people to have a Metabase account. Without a Metabase account, Metabase can't route the queries because it doesn't know who is viewing the embed.
```
